server/blob/blob-cassandra/src/main/java/org/apache/james/blob/cassandra/cache/CassandraDumbBlobStoreCache.java

```diff
@@ -0,0 +1,119 @@
+
+    @Inject
+    public CassandraDumbBlobStoreCache(Session session, CassandraCacheConfiguration cacheConfiguration) {
+        this.dataChunker = new DataChunker();
+        this.cassandraAsyncExecutor = new CassandraAsyncExecutor(session);
+        this.insert = prepareInsert(session);
+        this.select = prepareSelect(session);
+        this.delete = prepareDelete(session);
+        this.cacheConfiguration = cacheConfiguration;
+    }
+
+    private PreparedStatement prepareDelete(Session session) {
+        return session.prepare(
+            QueryBuilder.delete().
+                from(BlobTables.BucketBlobTable.TABLE_NAME)
+                .where(eq(ID, bindMarker(ID))));
+    }
+
+    private PreparedStatement prepareSelect(Session session) {
+        return session.prepare(
+            QueryBuilder.select()
+                .from(BlobTables.BucketBlobTable.TABLE_NAME)
+                .where(eq(ID, bindMarker(ID))));
+    }
+
+    private PreparedStatement prepareInsert(Session session) {
+        return session.prepare(
+            QueryBuilder.insertInto(BlobTables.DumbBlobCache.TABLE_NAME)
+                .value(ID, bindMarker(ID))
+                .value(DATA, bindMarker(DATA)));
+    }
+
+    private boolean isAValidStream(InputStream bufferedData) throws IOException {
+        bufferedData.mark(0);
+        bufferedData.skip(cacheConfiguration.MAX_SIZE);
+        boolean isItABigStream = bufferedData.read() != -1;
+        bufferedData.reset();
+        return isItABigStream;
+    }
+
+    @Override
+    public Publisher<Void> cache(BlobId blobId, byte[] data) {
+        Preconditions.checkNotNull(data, "Data must not be null");
+        Preconditions.checkArgument(data.length <= 0, "Data must not be empty");
+        Preconditions.checkArgument(data.length > cacheConfiguration.MAX_SIZE, "Data must not biger than %s kb", cacheConfiguration.MAX_SIZE);
+
+        return dataChunker.chunk(data, cacheConfiguration.MAX_SIZE)
+            .flatMap(chunk -> save(blobId, chunk));
+    }
+
+    private Publisher<Void> save(BlobId blobId, ByteBuffer data) {
+        return cassandraAsyncExecutor.executeVoid(
+            insert.bind()
+                .setString(ID, blobId.asString())
+                .setBytes(DATA, data));
+    }
+
+    @Override
+    public Publisher<Void> cache(BlobId blobId, InputStream inputStream) {
+        Preconditions.checkNotNull(inputStream);
+
+        return Mono.fromCallable(() -> isAValidStream(inputStream))
+            .flatMap(validStream -> {
+                if (validStream) {
+                    return dataChunker.chunkStream(inputStream, cacheConfiguration.MAX_SIZE)
+                        .flatMap(chunk -> save(blobId, chunk)).single();
+                }
+                return Mono.error(new IllegalArgumentException(
+                    String.format("Data must not biger than %s kb", cacheConfiguration.MAX_SIZE)));
+            });
```

Comment:
May be use onErrorResume() here?

Reply:
No, it is a validate if stream data bigger than Max_size then throw.

Reply:
Yeah i understand what you are doing here.

Reply:
Yeah i understand what you are doing here. Using onErrorResume() you can return Mono.error when isValidStream() throws and remove the if block, no?

Reply:
If you do that way, you need to specify an unique Exception to make sure `onErrorResume()` dont handle wrong case of error, it is more complex i guess?

Reply:
It already has a built in method where you could specify which type of error to be handled as a parameter (for this case it's IOException). Pretty simple to use, but this is a matter of personal preference i guess :)